src/storage/core.rs

```diff
@@ -1057,24 +1069,24 @@ where
         safe: &Safe<K>,
         key: &K,
         only_if_presented: bool,
-    ) -> Result<u64> {
+    ) -> Result<Option<DeleteResult>> {
```

Comment:
Looks like `Option` is redundant here. If there is no active BLOB, then `DeleteResult { dirty_bytes: 0, deleted: false }` can be returned. This way the code in `delete_core` will be a bit simpler